src/type/unit/Unit.js

```diff
@@ -1022,9 +1022,13 @@ export const createUnitClass = /* #__PURE__ */ factory(name, dependencies, ({
     // Replace this unit list with the proposed list
     ret.units = proposedUnitList
 
+    ret.value = null
```

Comment:
It took me a couple of cycles to understand why `ret.value = null` does not break anything (I expected it would). 

Is it an idea to move this line inside the new `if (this.value !== null) { ... } `? Then it is more clear that value is set to null so we can use it for the method `.to`.

Reply:
The `ret` variable will _always_ be a valueLess unit. It is either the valueLess unit in the `to(ret)` method to perform the correct unit conversion or it is returned directly if `this` is also valueLess.

I think you are correct that this line could be moved inside the `if (this.value !== null) { ... }` since that is the condition where `ret.value` is not already null from the clone. I'll make that change and test it. Thanks!

Reply:
👍 

Yes your code _is_ correctly working, it just took me a while to figure out why.

Reply:
Understood. There is no need to set it to `ret.value = null` in the case where it is already null. I've moved the line and everything still works fine.

Let me know if you have any other suggestions.

Reply:
Thanks 👌